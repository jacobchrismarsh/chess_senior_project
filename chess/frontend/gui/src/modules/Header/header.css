/* Style the header with a grey background and some padding */
.header {
  overflow: hidden;
  background-color: inherit;
}

/* Style the logo link (notice that we set the same value of line-height and font-size to prevent the header to increase when the font gets bigger */
.logo {
  font-size: 25px;
  font-weight: bold;
  color: black;
}

/* Float the link section to the right */
.header-right {
  float: right;
  padding: 10px;
}