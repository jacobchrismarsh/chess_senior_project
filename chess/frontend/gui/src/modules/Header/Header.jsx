import React, { Component } from "react";
import Button from '@material-ui/core/Button';
import autoBind from 'react-autobind';
import whiteKing from "../../cburnett/wk.svg";
import blackKing from "../../cburnett/bk.svg";
import blackQueen from "../../cburnett/bq.svg";
import whiteQueen from "../../cburnett/wq.svg";
import blackRook from "../../cburnett/br.svg";
import whiteRook from "../../cburnett/wr.svg";
import blackKnight from "../../cburnett/bn.svg";
import whiteKnight from "../../cburnett/wn.svg";
import blackBishop from "../../cburnett/bb.svg";
import whiteBishop from "../../cburnett/wb.svg";
import whitePawn from "../../cburnett/wp.svg";
import blackPawn from "../../cburnett/bp.svg";
import { Square } from "../Game/subcomponents/Square";
import "../Dashboard/dashboard.css";
import "./header.css";

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = { width: 0, height: 0 };
    autoBind(this);
  }

  handleHomeButton() {
    window.location = '/dashboard/';
  }

  handleSignOutButton() {
    localStorage.setItem('token', null);
    window.location = '/';
  }

  componentDidMount() {
    this.updateWindowDimensions();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }
  
  updateWindowDimensions() {
    this.setState({ width: window.innerWidth, height: window.innerHeight });
  }

  getRandomNumberOfSamePolarity(num, len) {
    let res = Math.trunc(Math.random() * len);
    while (this.isEven(num) !== this.isEven(res)) {
      res = Math.trunc(Math.random() * len);
    }

    return res;
  }

  isEven(num) {
    if (num === 0) {
      return true;
    }
    return num % 2 === 0;
  }

  renderBoarder() {
    let pieces = [whitePawn, blackPawn, whiteBishop, blackBishop, whiteKnight, blackKnight, whiteRook,
                  blackRook, whiteQueen, blackQueen, whiteKing, blackKing];
    let { width } = this.state;
    let widthOfPiece = 50;

    let renderElements = [];

    for (let i = 0; i < Math.floor(width / widthOfPiece) - 4; i++) {
      renderElements.push(<Square key={i} piece={pieces[this.getRandomNumberOfSamePolarity(i, pieces.length)]} />);
    }

    return (
      <div>
        {renderElements}
      </div>
    );
  }

  render() {
    return (
      <div className="header">
        {this.renderBoarder()}
        <div className="header-right">
          <Button
            size="medium"
            variant="contained"
            color="primary"
            className="chess-button"
            onClick={this.handleSignOutButton}
          >
            Sign Out
          </Button>
        </div>
        <div className="header-right">
          <Button
            size="medium"
            variant="contained"
            color="primary"
            className="chess-button"
            onClick={this.handleHomeButton}
          >
            Home
          </Button>
        </div>
      </div>
    );
  }
} 

export default Header;