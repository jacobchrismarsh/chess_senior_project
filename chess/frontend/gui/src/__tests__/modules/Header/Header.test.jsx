import React from 'react';
import { shallow } from 'enzyme';
import Header from "../../../modules/Header/Header";

describe("modules/Header/Header.jsx", () => {

  let header = null;

  beforeEach(() => {
    header = shallow(<Header />);
  });

  it("should render the Header component", () => {
    expect(header).toMatchSnapshot();
  });

  it("should return an even number when fed and even number", () => {
    let even = header.instance().getRandomNumberOfSamePolarity(2, 10);
    expect(even % 2 === 0).toBeTruthy();
  });

  it("should return an odd number when fed an odd number", () => {
    let odd = header.instance().getRandomNumberOfSamePolarity(1, 10);
    expect(odd % 2 !== 0).toBeTruthy();
  });

  it("should return true on an even number", () => {
    expect(header.instance().isEven(2)).toBeTruthy();
  });

  it("should return false on an odd number", () => {
    expect(header.instance().isEven(1)).not.toBeTruthy();
  })
});