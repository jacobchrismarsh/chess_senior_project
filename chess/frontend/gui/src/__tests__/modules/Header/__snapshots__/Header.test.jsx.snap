// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Header/Header.jsx should render the Header component 1`] = `
<div
  className="header"
>
  <div>
    <Square
      key="0"
      piece="wr.svg"
    />
    <Square
      key="1"
      piece="bq.svg"
    />
    <Square
      key="2"
      piece="wk.svg"
    />
    <Square
      key="3"
      piece="bb.svg"
    />
    <Square
      key="4"
      piece="wr.svg"
    />
    <Square
      key="5"
      piece="bb.svg"
    />
    <Square
      key="6"
      piece="wb.svg"
    />
    <Square
      key="7"
      piece="br.svg"
    />
    <Square
      key="8"
      piece="wr.svg"
    />
    <Square
      key="9"
      piece="bp.svg"
    />
    <Square
      key="10"
      piece="wp.svg"
    />
    <Square
      key="11"
      piece="bq.svg"
    />
    <Square
      key="12"
      piece="wn.svg"
    />
    <Square
      key="13"
      piece="bn.svg"
    />
    <Square
      key="14"
      piece="wn.svg"
    />
    <Square
      key="15"
      piece="br.svg"
    />
  </div>
  <div
    className="header-right"
  >
    <WithStyles(ForwardRef(Button))
      className="chess-button"
      color="primary"
      onClick={[Function]}
      size="medium"
      variant="contained"
    >
      Sign Out
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="header-right"
  >
    <WithStyles(ForwardRef(Button))
      className="chess-button"
      color="primary"
      onClick={[Function]}
      size="medium"
      variant="contained"
    >
      Home
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
